server/src/main/java/org/elasticsearch/transport/RemoteConnectionManager.java

```diff
@@ -102,11 +107,28 @@ public void openConnection(DiscoveryNode node, @Nullable ConnectionProfile profi
             node,
             profile,
             listener.delegateFailureAndWrap(
-                (l, connection) -> l.onResponse(wrapConnectionWithRemoteClusterInfo(connection, clusterAlias, credentialsManager))
+                (l, connection) -> l.onResponse(
+                    maybeLogDeprecationWarning(wrapConnectionWithRemoteClusterInfo(connection, clusterAlias, credentialsManager))
+                )
             )
         );
     }
 
+    private InternalRemoteConnection maybeLogDeprecationWarning(InternalRemoteConnection connection) {
+        if (connection.getClusterCredentials() == null
+            && (false == REMOTE_CLUSTER_PROFILE.equals(this.getConnectionProfile().getTransportProfile()))) {
+            deprecationLogger.warn(
+                DeprecationCategory.SECURITY,
+                "remote_cluster_certificate_access-" + connection.getClusterAlias(),
+                "The remote cluster connection to [{}] is using deprecated certificate based security model. "
```

Comment:
Grammar nit:

```suggestion
                "The remote cluster connection to [{}] is using the certificate-based security model. "
```

I think we need a "the" here but "the deprecated" sounds off, so I'd just skip "deprecated" since we mention it in the next sentence anyhow.